src/lesson2-homework.ts

```diff
@@ -1,109 +1,218 @@
+    }
+    return result;
   }
   // TODO: create an array of numbers
+  const nums: number[] = [1, 2, 3, 4, 5, 10]
   // TODO: create a function which doubles a number
+  const double = (num: number): number => num * 2
   // TODO: call map function (created earlier) with the array and the function
+  const doubled: number[] = map(nums, double)
   // TODO: print the result to console
+  console.log(`Exercise 7. Original array: ${nums}\nDoubled array: ${doubled}`)
 }
 // TODO: compile and run the code
 excercise7();
 
-// declare a function which takes a user and prits greeting to console
-function excercise8() {
+// declare a function which takes a user and prints greeting to console
+function excercise8(): void {
   // TODO: create a type for user, with name property
-  // TODO: create a function with name printGreeting, which takes a user and prits greeting to console
+  type TUser = {name: string};
+  // TODO: create a function with name printGreeting, which takes a user and prints greeting to console
+  const printGreeting = (user: TUser) => {
+    if (user && user.name && typeof user.name === 'string'){
+      console.log(`Exercise 8. Hello, ${user.name}`);
+    }else throw new Error('ArgumentType error')
+  }
   // TODO: create a type for product, with name property and price property
-  // TODO: create a product object, asign it some object literal
+  type TProduct = {name: string, price: number}
+  // TODO: create a product object, assign it some object literal
+  const phone: TProduct = {name: 'Iphone', price: 3000}
   // TODO: call the function with product as a parameter
+  printGreeting(phone)
   // TODO: call the function with object literal as a parameter
+  printGreeting({name: "Object Literal"})
   // TODO: try adding extra property to the object literal - observe the error
   // TODO: fix the error with type assertion
+  printGreeting({name: "Greg", lastname: "Agrot"} as TUser)
 }
 // TODO: compile and run the code
 excercise8();
 
 // declare a `Book` class with a constructor and a method
 function excercise9() {
   // TODO: declare a `Book` class with a constructor and a method `getInfo` which returns the book info as a string
-  // TODO: constructor should take three parameters - title and year of publication
-  // TODO: method `getInfo` should return the book title and year as a string
+  class Book {
+    // TODO: constructor should take three parameters - title and year of publication
+    protected title: string;
+    // TODO: add private modifier to the year property
+    // TODO: change protected modifier to the year property, remove private modifier
+    protected publicationYear: number
+    constructor(title: string, publicationYear: number) {
+      if (publicationYear > this.currentYear){
+        throw new Error(`Publication year could not be in the future - ${publicationYear}`)
+      }
+      this.title = title;
+      this.publicationYear = publicationYear;
+    }
+    private get currentYear(): number {
```

Comment:
in the real-world project this would likely be an util method, as it has no relation to the Book class

Reply:
I understand, would you recommend place it out of class?